networkx/algorithms/d_separation.py

```diff
@@ -359,6 +378,9 @@ def is_minimal_d_separator(G, u, v, z):
     if not nx.is_directed_acyclic_graph(G):
         raise nx.NetworkXError("graph should be directed acyclic")
 
+    if not nx.d_separated(G, {u}, {v}, z):
+        return False
+
```

Comment:
It looks like this call to `d_separated` calls `nx.is_directed_acyclic_graph` so we should remove the previous error check.

Also: The `union_uv` construction and error check on lines 384-389, are done inside the `d_separated` call. So those should be deleted too.

Reply:
Good catch. Removed.